src/main/java/org/threadly/litesockets/SocketExecuterCommonBase.java

```diff
@@ -56,73 +57,90 @@
     this.readScheduler = readScheduler;
     this.writeScheduler = writeScheduler;
   }
-
-  protected abstract void addThreadExecutorToClient(Client client);
-
-  @Override
-  public void addClient(final Client client) {
-    ArgumentVerifier.assertNotNull(client, "Client");
-    if(! client.isClosed() && client.getProtocol() == WireProtocol.TCP && isRunning()) {
-      addThreadExecutorToClient(client);
-      client.setClientsSocketExecuter(this);
-      if(client.getChannel() != null && client.getChannel().isConnected()) {
-        Client nc = clients.putIfAbsent(client.getChannel(), client);
-        if(nc == null) {
-          flagNewWrite(client);
-          flagNewRead(client);
-        }
-      } else {
-        client.connect();
-        if(client.getChannel() != null) {
-          Client nc = clients.putIfAbsent(client.getChannel(), client);
-          if(nc == null) {
-            readScheduler.execute(new AddToSelector(client, readSelector, SelectionKey.OP_CONNECT));
-            readSelector.wakeup();
-            dogCache.watch(client.connect(), client.getTimeout());
-          }
-        }
-      }
+  
+  protected void checkRunning() {
+    if(!isRunning()) {
+      throw new IllegalStateException("SocketExecuter is not running!");
     }
   }
-
+  
   @Override
-  public void removeClient(Client client) {
-    ArgumentVerifier.assertNotNull(client, "Client");
-    if(isRunning() && clients.remove(client.getChannel()) != null) {
-      readScheduler.execute(new RemoveFromSelector(client.getChannel(), readSelector));
-      writeScheduler.execute(new RemoveFromSelector(client.getChannel(), writeSelector));
-      readSelector.wakeup();
-      writeSelector.wakeup();
+  public TCPClient createTCPClient(final String host, final int port) throws IOException {
```

Comment:
make is two less key strokes than create...haha, but seriously, I do like this new API tons more.